### PR TITLE
Try to run tests on Windows Server

### PR DESCRIPTION
**Pull request by poncovka**
Do you know the one about hell and good intentions?

### CODE REVIEW

File: .travis/Dockerfile.windows-server [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM mcr.microsoft.com/windows/servercore:1809
+ENV LANG C.utf8
+
+# Install MSYS2. See: https://www.msys2.org/docs/ci/
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+RUN [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; \
+    Invoke-WebRequest -UseBasicParsing -uri "https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.sfx.exe" -OutFile msys2.exe; \
```
Comment (VladimirSlavik) [inline .travis/Dockerfile.windows-server:8]: I wonder about using msys2 nightly. Is that going to work well enough?

File: .travis/Dockerfile.windows-server [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    .\msys2.exe -y -oC:\; \
```
Comment (VladimirSlavik) [inline .travis/Dockerfile.windows-server:9]: Is renaming the installer on download a good idea? Using the same name as the project and probably some other executable does not seem too wise. You do specify `./msys2.exe` so it's probably enough, though.

File: .travis/Dockerfile.windows-server [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    Remove-Item msys2.exe ; \
+    C:\msys64\usr\bin\bash -lc ' '; \
+    C:\msys64\usr\bin\bash -lc 'pacman --noconfirm -Syuu'; \
+    C:\msys64\usr\bin\bash -lc 'pacman --noconfirm -Syuu'; \
```
Comment (VladimirSlavik) [inline .travis/Dockerfile.windows-server:13]: Why twice?

**Review by VladimirSlavik — COMMENTED** [review]
Some bits don't look right to me.

Then again, I have no idea about most of the pieces involved here, so... do as you see fit.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me and it is also fun :D.

**Comment by poncovka** [comment]
I am giving up on this one.

**Pull request closed without merging**